gradio/pipelines.py

```diff
@@ -3,268 +3,77 @@
-        pipeline,
-        pipelines.document_question_answering.DocumentQuestionAnsweringPipeline,  # type: ignore
-    ):
-        pipeline_info = {
-            "inputs": [
-                components.Image(type="filepath", label="Input Document", render=False),
-                components.Textbox(label="Question", render=False),
-            ],
-            "outputs": components.Label(label="Label", render=False),
-            "preprocess": lambda img, q: {"image": img, "question": q},
-            "postprocess": lambda r: {i["answer"]: i["score"] for i in r},
-        }
-    elif hasattr(transformers, "VisualQuestionAnsweringPipeline") and isinstance(
-        pipeline, pipelines.visual_question_answering.VisualQuestionAnsweringPipeline
-    ):
-        pipeline_info = {
-            "inputs": [
-                components.Image(type="filepath", label="Input Image", render=False),
-                components.Textbox(label="Question", render=False),
-            ],
-            "outputs": components.Label(label="Score", render=False),
-            "preprocess": lambda img, q: {"image": img, "question": q},
-            "postprocess": lambda r: {i["answer"]: i["score"] for i in r},
-        }
-    elif hasattr(transformers, "ImageToTextPipeline") and isinstance(
-        pipeline,
-        pipelines.image_to_text.ImageToTextPipeline,  # type: ignore
-    ):
-        pipeline_info = {
-            "inputs": components.Image(
-                type="filepath", label="Input Image", render=False
-            ),
-            "outputs": components.Textbox(label="Text", render=False),
-            "preprocess": lambda i: {"images": i},
-            "postprocess": lambda r: r[0]["generated_text"],
-        }
-    elif hasattr(transformers, "ObjectDetectionPipeline") and isinstance(
-        pipeline, pipelines.object_detection.ObjectDetectionPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Image(
-                type="filepath", label="Input Image", render=False
-            ),
-            "outputs": components.AnnotatedImage(
-                label="Objects Detected", render=False
-            ),
-            "preprocess": lambda i: {"inputs": i},
-            "postprocess": lambda r, img: (
-                img,
-                [
-                    (
-                        (
-                            i["box"]["xmin"],
-                            i["box"]["ymin"],
-                            i["box"]["xmax"],
-                            i["box"]["ymax"],
-                        ),
-                        i["label"],
-                    )
-                    for i in r
-                ],
-            ),
-        }
+    if str(type(pipeline)).startswith("<class 'transformers.pipelines"):
+        pipeline_info = _handle_transformers_pipeline(pipeline)
+
+    elif str(type(pipeline)).startswith("<class 'diffusers.pipelines"):
+        pipeline_info = _handle_diffusers_pipeline(pipeline)
```

Comment:
Relying on the `repr` seems a bit ugly, c.f. something like
```suggestion
    pipeline_class_name = pipeline.__class__.__name__
    if pipeline_class_name.startswith("transformers.pipelines"):
        pipeline_info = _handle_transformers_pipeline(pipeline)
        
    elif pipeline_class_name.startswith("diffusers.pipelines"):
        pipeline_info = _handle_diffusers_pipeline(pipeline)
```

Reply:
I know it looks ugly but I tried pipeline.__class__.__name__ and it just returns the class name lilke `TextClassificationPipeline`. It doesn't returns the whole path of class like type(). So I can't use your suggestion. I will still look what we can do.

Reply:
I found a better way, What do you say abut this ?? I tried multiple things out of which this looks the best
```suggestion
     if str(type(pipeline).__module__).startswith("transformers.pipelines"):
        pipeline_info = _handle_transformers_pipeline(pipeline)

    elif str(type(pipeline).__module__).startswith("diffusers.pipelines"):
        pipeline_info = _handle_diffusers_pipeline(pipeline)
```